apps/search_memory_index.cpp

```diff
@@ -67,15 +68,6 @@ int search_memory_index(diskann::Metric &metric, const std::string &index_path,
     }
 
     using TagT = uint32_t;
-    const bool concurrent = false, pq_dist_build = false, use_opq = false;
```

Comment:
Similarly to the other two tools, an argument can be made that this tool should continue to create the index here rather than move it to the `main` method (which mostly deals with parameters). Is there some angle that makes it cumbersome/not possible at this point?